File: codewars/Sum of Intervals/Sum_of_Intervals.cpp
#include <bits/stdc++.h>

using namespace std;

int sum_intervals(vector <pair <int, int>> intervals) {
    sort(intervals.begin(), intervals.end());
    int a = 0, L = intervals[0].first, R = intervals[0].second;
    for (int i = 1; i < intervals.size(); i++) {
        if (intervals[i].first >= R) {
            //a += (R + L - 1)*(R - L) / 2;
            a += R - L;
            L = intervals[i].first;
            R = intervals[i].second;
        }
        else {
            //a += (intervals[i].first + L - 1)*(intervals[i].first - L) / 2;
            a += intervals[i].first - L;
            L = intervals[i].first;
            R = max(R,intervals[i].second);
        }
    }
    //a += (R + L - 1)*(R - L) / 2;
    a += R - L;
    return a;
}

File: codewars/Sum of Intervals/sum of intervals.js
function sumIntervals(intervals) {
    const compareSort = (a, b) => a[0] - b[0];
    intervals.sort(compareSort);
    let prev;
    let intervalsTwo = [];
    intervalsTwo.push(intervals[0]);
    let ans = 0;
    for (let cur of intervals) {
        prev = intervalsTwo[intervalsTwo.length - 1];
        if (cur[0] <= prev[1]) {
            prev[1] = Math.max(prev[1], cur[1]);
        } else {
            intervalsTwo.push(cur);
        }
    }
    for (i of intervalsTwo) {
        ans += i[1] - i[0];
    }
    return ans
}